---
title: "Principal Component Analysis (PC-AiR)"
author: "Daniel Araujo"
date: "2020-11-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

To infer and analyze population structure, I performed PC-AiR at the autosomal genotype data. PC-AiR is a method used to perform a PCA accounting for relatedness among individuals, thus accurately capturing population structure.

## 1. Autosomal genotype data QC
```{r eval=FALSE}
plink --bfile MESA_subset --chr 1-22 --geno 0.01 --hwe 0.000001 --maf 0.05 --make-bed --out PCAir/MESA_auto_filt
```
The command line above was used to create new bim/bed/fam files containing genotype data corresponding to autosomal chromosomes (hence the ```--chr 1-22``` flag). Variants with missing call rates > 0.01 (```--geno 0.01```), HWE p-value < 10^-6 (```--hwe 0.000001```), and minor allele frequency < 0.05 (```--maf 0.05```) were removed. 

## 2. Performing PC-AiR
The following R script code was ran on the output created in the previous step. The script is a compiled file of four other R scripts written by Ryan and it's divided into four parts: i) conversion of PLINK files into GDS files, ii) estimation of kinship coefficient, iii) LD pruning, and iv) running PC-AiR. 

```{r eval=FALSE}
## Script to run PC-Air

# Loading libraries
library(GENESIS)
library(SNPRelate)
library(GWASTools)
library(dplyr)
library(tibble)
library(data.table)

# 01. Make GDS
snpgdsBED2GDS(bed.fn="MESA_auto_filt.bed",
              bim.fn="MESA_auto_filt.bim",
              fam.fn="MESA_auto_filt.fam",
              out.gdsfn="00autosome.gds")

# 02. King estimation
gdsfile<-"00autosome.gds"
gds <- snpgdsOpen(gdsfile)
king<-snpgdsIBDKING(gds)
kingMat<-king$kinship
snpgdsClose(gds)
saveRDS(king,file="King_matrix.RDS")

# 03. LD prune
gdsfile<-"00autosome.gds"
gds <- snpgdsOpen(gdsfile)
snpset <- snpgdsLDpruning(gds, method="corr", slide.max.bp=10e6,
                          ld.threshold=sqrt(0.3), verbose=FALSE)
pruned <- unlist(snpset, use.names=FALSE)
saveRDS(pruned,"pruned_set.RDS")
snpgdsClose(gds)

# 04. Run PCAIR
gdsfile<-"00autosome.gds"
pruned<-readRDS("pruned_set.RDS")
king<-readRDS("King_matrix.RDS")
kingMat<-king$kinship
colnames(kingMat)<-king$sample.id
row.names(kingMat)<-king$sample.id
geno <- GdsGenotypeReader(filename = gdsfile)
genoData <- GenotypeData(geno)
mypcair <- pcair(genoData, kinobj = kingMat, divobj = kingMat,
                 snp.include = pruned)
png("PCAIR_PC1_VS_PC2.png")
plot(mypcair)
dev.off()
eigenvec<-mypcair$vectors %>% as.data.frame() %>% rownames_to_column(var="sample_id")
str(eigenvec)
val<-mypcair$values %>% as.data.frame()
fwrite(eigenvec,"PCAIR.eigenvec",col.names = T,row.names = F,sep='\t')
fwrite(val,"PCAIR.eigenval")
```
PCA plots were generated to each individual population and to the full dataset. They are as follows:

![PC for the full dataset](assets/PCAIR_PC1_VS_PC2.png)
**PC Plot 1.** Full dataset
![PC for the AFA pop](assets/AFA_PCAIR_PC1_VS_PC2.png)
**PC Plot 2.** AFA pop
![PC for the CAU pop](assets/CAU_PCAIR_PC1_VS_PC2.png)
**PC Plot 3.** CAU pop
![PC for the CHN pop](assets/CHN_PCAIR_PC1_VS_PC2.png)
**PC Plot 4.** CHN pop
![PC for the HIS pop](assets/HIS_PCAIR_PC1_VS_PC2.png)
**PC Plot 5.** HIS pop