---
title: "Principal Component Analysis (PC-AiR)"
author: "Daniel Araujo"
date: "2020-12-08"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

To infer and analyze population structure, I performed PC-AiR at the autosomal genotype data. PC-AiR is a method used to perform a PCA accounting for relatedness among individuals, thus accurately capturing population structure.

## 1. Autosomal genotype data QC
```{r eval=FALSE}
plink --bfile MESA_subset --chr 1-22 --geno 0.01 --hwe 0.000001 --maf 0.05 --make-bed --out PCAir/MESA_auto_filt
```
The command line above was used to create new bim/bed/fam files containing genotype data corresponding to autosomal chromosomes (hence the `--chr 1-22` flag). Variants with missing call rates > 0.01 (```--geno 0.01```), HWE p-value < 10^-6 (```--hwe 0.000001```), and minor allele frequency < 0.05 (```--maf 0.05```) were removed. 

## 2. Separating populations
```{r eval=FALSE}
plink --bfile SHARE_MESA_c1 --keep /IDs_files/<pop>subset_ids.txt --make-bed --out MESA_subset_<pop>
```
To perform PC-AiR in each population, a set of PLINK files was created for each one (AFA, CAU, CHN, HIS). To do this I used the `--keep` flag followed by the path to a file containing the individuals' IDs - all .txt files used to do this step can be found on `/home/daniel/MESA_genotypes_subset/IDs_files`.

## 3. Performing PC-AiR
The following R script code was performed on the outputs created in the previous step. The script is a compiled file of four other R scripts written by Ryan and it's divided into four parts: i) conversion of PLINK files into GDS files, ii) estimation of kinship coefficient, iii) LD pruning, and iv) running PC-AiR. 

```{r eval=FALSE}
## Script to run PC-Air

# Loading libraries
library(GENESIS)
library(SNPRelate)
library(GWASTools)
library(dplyr)
library(tibble)
library(data.table)

# 01. Make GDS
snpgdsBED2GDS(bed.fn="MESA_auto_filt.bed",
              bim.fn="MESA_auto_filt.bim",
              fam.fn="MESA_auto_filt.fam",
              out.gdsfn="00autosome.gds")

# 02. King estimation
gdsfile<-"00autosome.gds"
gds <- snpgdsOpen(gdsfile)
king<-snpgdsIBDKING(gds)
kingMat<-king$kinship
snpgdsClose(gds)
saveRDS(king,file="King_matrix.RDS")

# 03. LD prune
gdsfile<-"00autosome.gds"
gds <- snpgdsOpen(gdsfile)
snpset <- snpgdsLDpruning(gds, method="corr", slide.max.bp=10e6,
                          ld.threshold=sqrt(0.3), verbose=FALSE)
pruned <- unlist(snpset, use.names=FALSE)
saveRDS(pruned,"pruned_set.RDS")
snpgdsClose(gds)

# 04. Run PCAIR
gdsfile<-"00autosome.gds"
pruned<-readRDS("pruned_set.RDS")
king<-readRDS("King_matrix.RDS")
kingMat<-king$kinship
colnames(kingMat)<-king$sample.id
row.names(kingMat)<-king$sample.id
geno <- GdsGenotypeReader(filename = gdsfile)
genoData <- GenotypeData(geno)
mypcair <- pcair(genoData, kinobj = kingMat, divobj = kingMat,
                 snp.include = pruned)
png("PCAIR_PC1_VS_PC2.png")
plot(mypcair)
dev.off()
eigenvec<-mypcair$vectors %>% as.data.frame() %>% rownames_to_column(var="sample_id")
str(eigenvec)
val<-mypcair$values %>% as.data.frame()
fwrite(eigenvec,"PCAIR.eigenvec",col.names = T,row.names = F,sep='\t')
fwrite(val,"PCAIR.eigenval")
```

## 4. Computing identity by descent (IBD)

To investigate relatedness among individuals in our dataset, we decided to compute identity by descent in each individual population. For this, the following code was used:

```{r eval=FALSE}
plink --bfile <pop>_MESA_autofilt --indep-pairwise 50 5 0.2 --out <pop>_pruned && plink --bfile <pop>_MESA_autofilt --extract <pop>_pruned.prune.in --genome --out <pop>_IBD_genome
```

By using the command line above, we firstly perform a LD-based SNP pruning using `--indep-pairwise` followed by the window size in Kb, step size, and variance inflation factor. Then, using the output of the previous command, we perform IBD estimation using the `--genome` flag.

## 5. PCA, Scree, and IBD plots

For better visualization, the PCA plot for all populations together was colored in R by population using the following code:

```{r eval=FALSE}
library(ggplot2) #loading ggplot
eigenvec <- read.table("updated_eigenvec.txt", header=T) #reads the eigenvec file
ggplot(eigenvec, aes(x=V1,y=V2,color=pop)) + geom_point() +
  labs(title="PC1 vs PC2") #plots graph
```

Scree plots showing how much of the variation each PC accounts for were made using the following R code:

```{r eval=FALSE}
library(ggplot2) #loading ggplot
eigenval <- read.table("<pop>_PCAIR.eigenval", header=T, nrows=10) #reads the first 10 rows of the eigenval file
eigenval$variation_perc <- eigenval$./sum(eigenval$.) #creates a new column in which absolute values are now proportions
sum(eigenval$variation_perc) #double-check to see if the new column adds up to 1
ggplot(eigenval, aes(x=c(1:10),y=variation_perc)) + geom_point() +
  geom_line() + xlab("PC") + ylab("Variation") + 
  scale_x_continuous(breaks = 1:10) +
  labs(title="Proportion of variance explained") #plots graph
```

IBD plots were generated for each population using the following R code:

```{r eval=FALSE}
library(ggplot2) #loading ggplot
<pop>_ibd <- read.table("<pop>_IBD_genome.genome", header=T) #reads .genome file
ggplot(AFA_ibd, aes(x=Z0,y=Z1,color=RT)) + geom_point() + xlim(0:1) + ylim(0:1) #plots graph
```

Plots are as follows:

### ALL

![1. PCA plot for ALL pops](assets/ALL_PCA_colored_pop.png)

![2. Variation explained by the first 10 PCs](assets/ALL_PCprop.png)

***

### AFA

![3. PCA plot for AFA pop](assets/AFA_PCAIR_PC1_VS_PC2.png)

![4. Variation explained by the first 10 PCs](assets/AFA_PCprop.png)

![5. IBD plot for AFA pop](assets/AFA_IBD.png)


***

### CAU

![6. PCA plot for CAU pop](assets/CAU_PCAIR_PC1_VS_PC2.png)

![7. Variation explained by the first 10 PCs](assets/CAU_PCprop.png)

![8. IBD plot for CAU pop](assets/CAU_IBD.png)

***

### CHN

![9. PCA plot for CHN pop](assets/CHN_PCAIR_PC1_VS_PC2.png)

![10. Variation explained by the first 10 PCs](assets/CHN_PCprop.png)

![11. IBD plot for CHN pop](assets/CHN_IBD.png)


***

### HIS

![12. PCA plot for HIS pop](assets/HIS_PCAIR_PC1_VS_PC2.png)

![13. Variation explained by the first 10 PCs](assets/HIS_PCprop.png)

![14. IBD plot for HIS pop](assets/HIS_IBD.png)
